pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -55,12 +80,140 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 			}
 		}
 	}
+	return nil
+}
 
-	instance.Status.StorageClassesCreated = true
-
+// ensureExternalCluster ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalCluster(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalClusterDetailsSecret,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ExternalResource
+	err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
+	if err != nil {
+		reqLogger.Error(err, "could not parse json blob")
+		return err
+	}
+	scs, err := r.newStorageClasses(instance)
+	if err != nil {
+		return err
```

Comment:
name your error

Reply:
I'm not quite sure what you meant here?. 
So instead of using `err` I should use so something like `newStorageClasserr`?.

Reply:
I told you not to return `err` but explain the error, somewhere else you logged and returned `err` not sure who recommended that but go ahead and follow that pattern

Reply:
`newStorageClasses()` should be responsible for elaborating on its own errors, we just need to pass it up. This is outside the scope of this PR.